diff --git a/server/test/cats.spec.ts b/server/test/cats.spec.ts
index 5636332..6a4b64e 100644
--- a/server/test/cats.spec.ts
+++ b/server/test/cats.spec.ts
@@ -10,42 +10,44 @@ const newCat = { name: 'Fluffy', weight: 4, age: 2 };
 let catId = '';
 
 beforeAll(async () => {
+  await app.ready();
   await connectToMongo();
 });
 
 afterAll(async () => {
-  await request(app).delete('/api/cats/delete');
+  await request(app.server).delete('/api/cats/delete');
+  await app.close();
   await disconnectMongo();
 });
 
 describe('Cat tests', () => {
   test('should get all cats', async () => {
-    const res = await request(app).get('/api/cats');
+    const res = await request(app.server).get('/api/cats');
     expect(res.statusCode).toBe(200);
     expect(res.body).toStrictEqual([]);
   });
   test('should count all cats', async () => {
-    const res = await request(app).get('/api/cats/count');
+    const res = await request(app.server).get('/api/cats/count');
     expect(res.statusCode).toBe(200);
     expect(res.body).toStrictEqual(0);
   });
   test('should create a new cat', async () => {
-    const res = await request(app).post('/api/cat').send(newCat);
+    const res = await request(app.server).post('/api/cat').send(newCat);
     expect(res.statusCode).toBe(201);
     expect(res.body).toMatchObject(newCat);
     catId = res.body._id;
   });
   test('should get a cat by id', async () => {
-    const res = await request(app).get(`/api/cat/${catId}`);
+    const res = await request(app.server).get(`/api/cat/${catId}`);
     expect(res.statusCode).toBe(200);
     expect(res.body).toMatchObject(newCat);
   });
   test('should update a cat by id', async () => {
-    const res = await request(app).put(`/api/cat/${catId}`).send({ weight: 5 });
+    const res = await request(app.server).put(`/api/cat/${catId}`).send({ weight: 5 });
     expect(res.statusCode).toBe(200);
   });
   test('should delete a cat by id', async () => {
-    const res = await request(app).delete(`/api/cat/${catId}`);
+    const res = await request(app.server).delete(`/api/cat/${catId}`);
     expect(res.statusCode).toBe(200);
   });
 });
diff --git a/server/test/users.spec.ts b/server/test/users.spec.ts
index e60c602..b101bad 100644
--- a/server/test/users.spec.ts
+++ b/server/test/users.spec.ts
@@ -10,42 +10,44 @@ const newUser = { username: 'Dave', email: 'dave@example.com', role: 'user' };
 let userId = '';
 
 beforeAll(async () => {
+  await app.ready();
   await connectToMongo();
 });
 
 afterAll(async () => {
-  await request(app).delete('/api/users/delete');
+  await request(app.server).delete('/api/users/delete');
+  await app.close();
   await disconnectMongo();
 });
 
 describe('User tests', () => {
   test('should get all users', async () => {
-    const res = await request(app).get('/api/users');
+    const res = await request(app.server).get('/api/users');
     expect(res.statusCode).toBe(200);
     expect(res.body).toStrictEqual([]);
   });
   test('should count all users', async () => {
-    const res = await request(app).get('/api/users/count');
+    const res = await request(app.server).get('/api/users/count');
     expect(res.statusCode).toBe(200);
     expect(res.body).toStrictEqual(0);
   });
   test('should create a new user', async () => {
-    const res = await request(app).post('/api/user').send(newUser);
+    const res = await request(app.server).post('/api/user').send(newUser);
     expect(res.statusCode).toBe(201);
     expect(res.body).toMatchObject(newUser);
     userId = res.body._id;
   });
   test('should get a user by id', async () => {
-    const res = await request(app).get(`/api/user/${userId}`);
+    const res = await request(app.server).get(`/api/user/${userId}`);
     expect(res.statusCode).toBe(200);
     expect(res.body).toMatchObject(newUser);
   });
   test('should update a user by id', async () => {
-    const res = await request(app).put(`/api/user/${userId}`).send({ username: 'Dave2' });
+    const res = await request(app.server).put(`/api/user/${userId}`).send({ username: 'Dave2' });
     expect(res.statusCode).toBe(200);
   });
   test('should delete a user by id', async () => {
-    const res = await request(app).delete(`/api/user/${userId}`);
+    const res = await request(app.server).delete(`/api/user/${userId}`);
     expect(res.statusCode).toBe(200);
   });
 });
